RELEASE_PROCESS.md

```diff
@@ -24,7 +24,7 @@ git push origin release-0.3.1
 
 ## If something goes wrong
 
-The release process can be restarted at various points in it's lifecycle by pushing specially crafted tags.
+The release process can be restarted at various points in it's life-cycle by pushing specially crafted tags.
```

Comment:
While you're in here: `it's` => `its`
And `lifecycle` is at least as valid as `life-cycle`.